File: piotr/org.uva.sc.piotr.qls/src/main/java/qls/ast/model/Stylesheet.java
package qls.ast.model;

import qls.ast.visitors.ASTNodeVisitor;

import java.util.List;

public class Stylesheet extends ASTNode {
    private final String name;
    private final List<Page> pages;

    public Stylesheet(MetaInformation metaInformation, String name, List<Page> pages) {
        super(metaInformation);
        this.name = name;
        this.pages = pages;
    }

    @Override
    public <T> T accept(ASTNodeVisitor<T> visitor) {
        return visitor.visit(this);
    }

    public String getName() {
        return name;
    }

    public List<Page> getPages() {
        return pages;
    }

}

File: DogeQL/src/main/kotlin/qls/ast/node/StyleSheet.kt
package qls.ast.node

import qls.common.Name
import qls.visitor.QlsVisitor

data class StyleSheet(val pages: List<Page>, val name: Name) : QlsNode {
    override fun <T> accept(visitor: QlsVisitor<T>): T {
        return visitor.visit(this)
    }
}